audio/xmms2/Portfile

```diff
@@ -52,6 +53,12 @@ depends_lib-append      path:lib/pkgconfig/glib-2.0.pc:glib2 \
                         port:libvorbis \
                         port:wavpack
 
+# Build system does not provide per-component configure args.
+# By default everything is used, w/e is found.
+# A list of components to disable can be passed to configure
+# via --without-plugins=, but it seems easier to use conflicts_build.
+conflicts_build         jack pulseaudio
```

Comment:
`conflicts_build` should be a last resort because it's pretty inconvenient for users, especially when they're trying to automate port installations or updates; see for example [this mailing list post](https://lists.macports.org/pipermail/macports-dev/2024-September/045850.html) where someone complained about needing to follow the instructions from `conflicts_build` yesterday. If `--without-plugins=` works to avoid using those dependencies, let's do that instead.

Reply:
@ryandesign It probably will, but we need to handle three cases (no jack no pulse, only jack, only pulse), and I am not sure how to do this nearly. Portfile already has some hackery for optionals.

If you can suggest how to implement it nicely, then sure.

Reply:
Yes, I would handle plugins the same way the port handles optionals. 

I think the optionals code can first be simplified slightly: `--with-optionals=` doesn't need to be in the initial `configure.args` and in the `pre-configure` block, `--with-optionals=[join ${optionals} ,]` can just be appended to `configure.args`. If `${optionals}` is empty that should still be fine.

Reply:
I have probably fixed this now.